go-controller/pkg/informer/informer.go

```diff
@@ -26,6 +28,16 @@ type EventHandler interface {
 	Synced() bool
 }
 
+var underTest bool = false
+
+func init() {
```

Comment:
This is pretty terrible... Can you describe in the commit message for this commit why we need to skip the version check?

Reply:
Yes it's awful 🙀 
I'll adjust the commit message for sure, but per the PR description:

```
The informer package won't process updates under test conditions when the informers have been started. This is due to a check in to ensure the objects K8s revision number has incremented and that therefore this is genuine work that we want to enqueu - that revision number isn't implemented in the fakeClient. Rather than implement this, I added a condition to detect when being run under test and skip this check.
```

Reply:
if this is filtering all the updates how are the test flaky? I mean, they should fail always on all the test that expect actions on objects that are being updated

Reply:
Ha. You raise a valid point @aojea.

The tests were only passing because the test logic was racing against the informer startup🤪

As soon as I added `f.WaitForCacheSync`, a the flaky tests failed 100% of the time.

In other words, if the informer cache reached sync *before* the main body of the test started, then we'd see a test failure as any cache events that came afterwards were ignored (due to the limitation of `fakeClient` explained above).

Reply:
Also, people smarter than me (i.e people who wrote example on how to use workqueues, the folks that wrote controller-runtime, and the kubehandler authors) all have variants of this check - which to me signals that it's likely to cause us unnecessary work in the real world.

Therefore, as nasty as it is, I'd prefer to conditionally drop the check for tests rather than remove it entirely

Reply:
@dave-tucker could we instead add a new Informer creator function that ignores the versioning internally? eg http://people.redhat.com/dcbw/testinformer.patch

Reply:
@dcbw I took inspiration from your patch and went a slightly different route to achieve the same objective.
basically, I passed the function for creating EventHandlers in to NewNode, NewMaster etc...
This is tidier IMHO as we don't leak implementation details of the informer code (in this case, whether or not we do the ResourceVersion check) in to the Master and Node structs.